osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -33,9 +33,9 @@ public Strain(Mod[] mods, int totalColumns)
             overallStrain = 1;
         }
 
-        protected override double StrainValueOf(DifficultyHitObject current)
+        protected override double StrainValueOf(DifficultyHitObjectIterator iterator)
```

Comment:
I don't really want to push my idea further as I'm not wholly sold on it myself (which is why I mentioned it as an off-hand remark), but do you have examples for how this takes "legibility one step backwards" rather than just mentioning that without any supporting reasoning?

I'd take this diff any day over some diffs I've read in the past that had `lastObject` and `lastLastObject` and `fourthLastObject` and whatever. Maybe the name can be worked on, sure, but a vague mention of "legibility" is not really addressable in any way shape or form.

Reply:
I mean, having diffcalc methods like `StrainValueOf`, `CalculateInitialStrain`, or even `Skill.Process` accepting an iterator reads really weirdly, that's what I mean by "legibility" here. 

I'm not sure if you have any better term for that, but I would much rather have a list provided by `DifficultyHitObject` rather than an "iterator" wrapper replacing `DifficultyHitObject` everywhere.

Reply:
I would have presumed that storing every single `DifficultyHitObject` as a list in each `DifficultyHitObject` wouldn't be great in terms of allocs but I could be wrong? Would be interested in finding out 👀 

As for legibility, I can agree that it is less intuitive from a pure naming standpoint... I'd personally be fine making the sacrifice here though, and the whole difficulty calculation flow is currently subject to change anyway

Reply:
> I would have presumed that storing every single DifficultyHitObject as a list in each DifficultyHitObject wouldn't be great in terms of allocs but I could be wrong? Would be interested in finding out 👀

The idea is to provide the full list inside each `DifficultyHitObject` only as a reference, for usages to figure out the objects behind and ahead of the current one. Only providing this as an alternative suggestion, others may provide better ideas to this.